surefire-its/src/test/resources/surefire-1993-jpms-providing-modules/application/pom.xml

```diff
@@ -0,0 +1,94 @@
+~ KIND, either express or implied.  See the License for the
+~ specific language governing permissions and limitations
+~ under the License.
+-->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>org.apache.maven.plugins.surefire</groupId>
+        <artifactId>surefire-1993-jpms-providing-modules</artifactId>
+        <version>1.0-SNAPSHOT</version>
+    </parent>
+
+    <groupId>org.apache.plugins.failsafe.its</groupId>
+    <artifactId>application</artifactId>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.plugins.failsafe.its</groupId>
+            <artifactId>studentservice</artifactId>
+            <version>1.0-SNAPSHOT</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.plugins.failsafe.its</groupId>
+            <artifactId>studentservice-provider</artifactId>
+            <version>1.0-SNAPSHOT</version>
+        </dependency>
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-api</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-engine</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.assertj</groupId>
+            <artifactId>assertj-core</artifactId>
+        </dependency>
+    </dependencies>
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <configuration>
+                    <argLine>
+                        --add-modules ALL-DEFAULT
+                        --add-opens application/org.apache.failsafe.its.serviceloader.application=ALL-UNNAMED
+                    </argLine>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-failsafe-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <goals>
+                            <goal>integration-test</goal>
+                            <goal>verify</goal>
+                        </goals>
+                    </execution>
+                </executions>
+                <configuration>
+                    <argLine>
+                        --add-modules ALL-DEFAULT
+                        --add-opens application/org.apache.failsafe.its.serviceloader.application=ALL-UNNAMED
+                    </argLine>
+                    <forkCount>1</forkCount>
+                    <forkMode>once</forkMode>
```

Comment:
These two parameters are not necessary because they have default values here.

Reply:
When I remove them, the test fails with

```
[WARNING] The parameter forkMode is deprecated since version 2.14. Use forkCount and reuseForks instead.
[WARNING] useSystemClassLoader setting has no effect when not forking
[WARNING] The parameter forkCount should likely not be 0. Forking a JVM for tests improves test accuracy. Ensure to have a <forkCount> >= 1.
```

I guess this means that Failsafe is not forking a JVM and hence not setting the module path, could that be the case?

Reply:
Ah ok, I know what you mean. It's because of the [parent](https://github.com/apache/maven-surefire/blob/master/surefire-its/src/test/resources/pom.xml#L23). It would be nice to remove these parameters but nobody has spare time to make impact analysis in those ITs wjich inherit this parent and move these parameters into particular target POM of the ITs. If you like, you can keep it in this PR as it is and open another PR where all such cases will be fixed.

```
    <parent>
        <groupId>org.apache.maven.surefire</groupId>
        <artifactId>it-parent</artifactId>
        <version>1.0</version>
        <relativePath>../pom.xml</relativePath>
    </parent>
```